core/drivers/firewall/firewall.c

```diff
@@ -90,6 +176,57 @@ TEE_Result firewall_set_configuration(struct firewall_query *fw)
 	return fw->ctrl->ops->set_conf(fw);
 }
 
+TEE_Result firewall_set_configuration(struct firewall_conf *conf)
+{
+	TEE_Result res = TEE_ERROR_GENERIC;
+	size_t i = 0;
+
+	assert(conf && conf->nb_queries);
+
+	for (i = 0; i < conf->nb_queries; i++) {
+		/*
+		 * In case of error, report it to the caller. Note that
+		 * the firewall configurations may be partially loaded. In such
+		 * case, it is the caller responsibility to decide what to do.
```

Comment:
This description should be also found in `firewall_set_configuration()` inline description comment.